diff --git a/Config/Makefile.Linux-2-6-24 b/Config/Makefile.Linux-2-6-24
index 58e0608..01d80ab 100644
--- a/Config/Makefile.Linux-2-6-24
+++ b/Config/Makefile.Linux-2-6-24
@@ -28,7 +28,8 @@ SYSDEFS =
 CPPFLAGS = -I. 
 #0 INCLS = -I. @V_INCL@
 DEFS = -DHAVE_CONFIG_H
-LIBS = -lwrap -lutil  -lreadline  -lssl
+LIBS = -L${libdir} 
+Libs.private: -lz -pthread -lm -lutil -lreadline -lssl
 LDFLAGS = 
 
 INSTALL = /usr/bin/install -c

diff --git a/configure.in b/configure.in
index 45cb88a..581ba04 100644
--- a/configure.in
+++ b/configure.in
@@ -1540,7 +1540,7 @@ if test -n "$WITH_LIBWRAP" -a "$sc_cv_have_tcpd_h" = yes; then
       if test -n "$LIBWRAP_ROOT"; then
        L="$LIBWRAP_ROOT/lib"; LIBS="-L$L -lwrap $LIBS"
       else
-       LIBS="-lwrap $LIBS"
+       LIBS="$LIBS"
       fi
       AC_TRY_LINK([#include <sys/types.h>
 #include <tcpd.h>
